packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,288 @@
+import { sortable } from "@patternfly/react-table";
+import {
+  BucketClassType,
+  DATA_FEDERATION_NAMESPACE,
+  NOOBAA_TYPE_MAP
+} from "../../constants";
+import {
+  NooBaaBackingStoreModel,
+  NooBaaBucketClassModel,
+  NooBaaNamespaceStoreModel
+} from "../../models";
+import { BucketClassKind, NamespaceStoreKind } from "../../types";
+import { referenceForModel } from "../../utils";
+import { OperandStatus } from "../../utils/generics/operand-status";
+import { Kebab } from "../../utils/kebab/kebab";
+import {
+  LaunchModal,
+  ModalMap,
+  useModalLauncher
+} from "../../utils/modals/modalLauncher";
+import { getName } from "../../utils/selectors/k8s";
+import "./list-page.scss";
+
+const tableColumnInfo = [
+  { className: "", id: "name" },
+  {
+    className: "",
+    id: "status"
+  },
+  {
+    className: "",
+    id: "provider"
+  },
+  {
+    className: "",
+    id: "buckets"
+  },
+  { className: "dropdown-kebab-pf pf-c-table__action", id: "" }
+];
+
+type ResourceTableProps = {
+  data: K8sResourceCommon[];
+  unfilteredData: K8sResourceCommon[];
+  loaded: boolean;
+  loadError: any;
+  rowData: CustomData;
+};
+
+const ResourceTable: React.FC<ResourceTableProps> = (props) => {
+  const { t } = useTranslation();
+  const tableColumns = React.useMemo<TableColumn<K8sResourceCommon>[]>(
+    () => [
+      {
+        title: t("Name"),
+        sort: "metadata.name",
+        transforms: [sortable],
+        props: {
+          className: tableColumnInfo[0].className
+        },
+        id: tableColumnInfo[0].id
+      },
+      {
+        title: t("Status"),
+        props: {
+          className: tableColumnInfo[1].className
+        },
+        id: tableColumnInfo[1].id
+      },
+      {
+        title: t("Provider"),
+        props: {
+          className: tableColumnInfo[2].className
+        },
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const resource = obj as K8sResourceCommon & {
+    provider?: string;
+    buckets?: number;
```

Comment:
let's add this in `rowData` only instead of `obj`. `obj` is something which is used by other components as well, there is no need to pass down `provider` and `buckets` info to any other components. `rowData` is what we have to use.